agreement_repair/views/agreement_view.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="action_mrp_repair_agreement_specific" model="ir.actions.act_window">
+        <field name="name">Repair Orders</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">repair.order</field>
+        <field name="view_mode">list,form</field>
+        <field name="domain">[('agreement_id', '=', active_id)]</field>
+        <field name="context">{'create': False, 'edit': False}</field>
+        <field name="help" type="html">
+            <p class="oe_view_nocontent_create">
+                Create Repair Orders
+            </p>
+        </field>
+    </record>
+
+    <record id="partner_agreement_form_view_repair" model="ir.ui.view">
+        <field name="name">agreement.form.repair</field>
+        <field name="model">agreement</field>
+        <field name="inherit_id" ref="agreement.agreement_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button
+                    name="%(action_mrp_repair_agreement_specific)d"
+                    type="action"
+                    class="oe_stat_button"
+                    icon="fa-wrench"
```

Comment:
```suggestion
                    icon="fa-wrench"
                    invisible="repair_count == 0"
```

For better UX, aligned to how Odoo handles this kind of things
e.g.: https://github.com/odoo/odoo/blob/c84b67e5dca1a74e1a5862e754455944015f4105/addons/sale/views/sale_order_views.xml#L328